src/main/scala/org/apache/spark/shuffle/ucx/NvkvShuffleMapOutputWriter.scala

```diff
@@ -0,0 +1,291 @@
+        partChannel = new PartitionWriterChannel(partitionId)
+      }
+      Optional.of(partChannel)
+    }
+
+    override def getNumBytesWritten: Long = {
+      NvkvShuffleMapOutputWriter.log.debug("NvkvShufflePartitionWriter getNumBytesWritten " + partitionId)
+      if (partChannel != null) try partChannel.getCount
+      catch {
+        case e: IOException =>
+          throw new RuntimeException(e)
+      }
+      else if (partStream != null) partStream.getCount
+      else {
+        // Assume an empty partition if stream and channel are never created
+        0
+      }
+    }
+  }
+
+  private class PartitionWriterStream private[ucx](private val partitionId: Int) extends OutputStream {
+    NvkvShuffleMapOutputWriter.log.debug("PartitionWriterStream ctor " + partitionId)
+    private var count = 0
+    private var isClosed = false
+
+    def getCount: Long = count
+
+    @throws[IOException]
+    override def write(b: Int): Unit = {
+      NvkvShuffleMapOutputWriter.log.debug("PartitionWriterStream write1 " + b)
+      verifyNotClosed()
+      count += 1
+    }
+
+    @throws[IOException]
+    override def write(buf: Array[Byte], pos: Int, length: Int): Unit = {
+      val offset = currChannelPosition + count
+      NvkvShuffleMapOutputWriter.log.debug(s"PartitionWriterStream write2 $shuffleId,$mapId,$partitionId buf $buf pos $pos length $length offset $offset")
+      nvkvWrapper.write(dsIdx, shuffleId, mapId, partitionId, buf, length, offset)
+      verifyNotClosed()
+      count += length
+    }
+
+    override def close(): Unit = {
+      var padding: Int = nvkvWrapper.writeRemaining(dsIdx, currChannelPosition+count)
+      nvkvWrapper.commitPartition(dsIdx, currChannelPosition, count, shuffleId, mapId, partitionId)
+      isClosed = true
+      partitionLengths(partitionId) = count
+      totalPartitionsPadding += padding
+      partitionsPadding(partitionId) = padding
+      bytesWrittenToMergedFile += count
+      NvkvShuffleMapOutputWriter.log.debug(s"PartitionWriterStream close currChannelPosition ${currChannelPosition}")
+      NvkvShuffleMapOutputWriter.log.debug(s"PartitionWriterStream close $shuffleId,$mapId,$partitionId count $count padding $padding bytesWrittenToMergedFile $bytesWrittenToMergedFile")
+    }
+
+    private def verifyNotClosed(): Unit = {
+      if (isClosed) throw new IllegalStateException("Attempting to write to a closed block output stream.")
+    }
+  }
+
+  private class PartitionWriterChannel private[ucx](private val partitionId: Int) extends WritableByteChannelWrapper {
+    NvkvShuffleMapOutputWriter.log.debug("PartitionWriterChannel ctor " + partitionId)
+
+    @throws[IOException]
+    def getCount: Long = {
+      NvkvShuffleMapOutputWriter.log.debug("PartitionWriterChannel getCount " + partitionId)
+      0
+    }
+
+    override def channel: WritableByteChannel = {
+      NvkvShuffleMapOutputWriter.log.debug("PartitionWriterChannel channel " + partitionId)
+      null
+    }
```

Comment:
Wouldn't it be better to declare those as variables?

Reply:
declare what?

Reply:
`getCount` and `channel`?

Reply:
Maybe I'm misunderstanding, but we can just use the count attribute, no?